File: itb1704-2022-track.io-main/BackEnd/Model/MealPlan.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

namespace BackEnd.Model
{
    [Table("mealplan")]
    public class MealPlan
    {
        [Column("id")]
        public int Id { get; init; }
        [Column("name")]
        public string? Name { get; init; }
        [Column("lenghtInDays")]
        public int LenghtInDays { get; init; }
        [Column("selected")]
        public bool Selected { get; set; }
        [Column("totalCals")]
        public int TotalCals { get; set; }
        [Column("calsInDay")]
        public int CalsInDay { get; set; }

        [Column("mealPlanNutritions")]
        [JsonIgnore]
        public virtual ICollection<MealPlanNutrition> MealPlanNutritions { get; init; } = new List<MealPlanNutrition>();
    }
}

File: itb1704-2022-track.io-main/FrontENd/src/modules/mealplan.ts

import { MealPlanNutrition } from "./mealPlanNutrition";

export interface MealPlan {
    id: number;
    name?: string;
    lenghtInDays: number;
    mealPlanNutritions: MealPlanNutrition;
    selected: boolean;
    totalCals: number;
    calsInDay:number;
  }